src/target/stm32h7.c

```diff
@@ -343,25 +371,55 @@ static bool stm32h7_flash_unlock(target_s *const target, const uint32_t addr)
 	return !(target_mem_read32(target, regbase + FLASH_CR) & FLASH_CR_LOCK);
 }
 
+/* Helper for offsetting FLASH_CR bits correctly */
+static uint32_t stm32h7_flash_cr(uint32_t sector_size, const uint32_t ctrl, int snb)
+{
+	uint32_t command = ctrl;
+	/* H74x, H72x IP: 128 KiB and has PSIZE */
+	if (sector_size == FLASH_SECTOR_SIZE) {
+		command = ctrl | (snb * FLASH_CR_SNB_1);
+		DEBUG_TARGET("%s: patching FLASH_CR from 0x%08" PRIx32 " to 0x%08" PRIx32 "\n", __func__, ctrl, command);
+		return command;
+	}
+
+	/* H7Bx IP: 8 KiB and no PSIZE */
+	/* Save and right-shift FW, START bits */
+	const uint32_t temp_fw_start = command & (FLASH_CR_FW | FLASH_CR_START);
+	/* Parallelism is ignored */
+	command &= ~(FLASH_CR_PSIZE64 | FLASH_CR_FW | FLASH_CR_START);
+	/* Restore FW, START to H7Bx-correct bits */
+	command |= (temp_fw_start >> 2U);
```

Comment:
The parens here are unnecessary, please drop them.

Reply:
Dropped a few parens. I don't get any highlighting for extraneous parens, so I may add more than what's enough per C rules (and `make clang-tidy` sucks with all the noise from headers)